controller/nodegroup.go

```diff
@@ -253,21 +253,34 @@ func createNodeGroup(config *eksv1.EKSClusterConfig, group eksv1.NodeGroup, eksS
 		nodeGroupCreateInput.Subnets = aws.StringSlice(config.Status.Subnets)
 	}
 
-	finalTemplate := fmt.Sprintf(templates.NodeInstanceRoleTemplate, getEC2ServiceEndpoint(config.Spec.Region))
-	output, err := createStack(svc, fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName), config.Spec.DisplayName, finalTemplate, []string{cloudformation.CapabilityCapabilityIam}, []*cloudformation.Parameter{})
-	if err != nil {
-		return "", err
+	generatedNodeRole := config.Status.GeneratedNodeRole
+
+	if aws.StringValue(group.NodeRole) == "" {
+		if config.Status.GeneratedNodeRole == "" {
```

Comment:
`config.Status.GeneratedRole` is not set until after the first reconciliation, so every node group will end up creating a stack. Is that by design?